### PR TITLE
Add security context for the cronjob, main container, main pod, and ngnix container

### LINKED ISSUE #186: Possible wrong extraSecurityContext
In `values.yaml` you have `nextcloud.extraSecurityContext` but it's not used anywhere in the helm chart.

https://github.com/nextcloud/helm/blob/master/charts/nextcloud/values.yaml#L174

What is used is `.securityContext`

https://github.com/nextcloud/helm/blob/master/charts/nextcloud/templates/deployment.yaml#L445

### LINKED ISSUE #252: securityContext is used both on pod and container level
The securityContext settings are used on a pod level in the `deployment.yaml` and on container level in the `cronjob.yaml`. Unfortunately those secuirtyContexts are not compatible. I have to set `securityContext.fsGroupChangePolicy` for the nextcloud deployment, but that setting does not exist on container level and therefore the Helm deployment fails.

Maybe it would make sense to split the security context for the nextcloud deployment and the cronjob?

### PR DESCRIPTION
**Pull request by Timmy1e**
# Pull Request

## Description of the change

Allows users to set the securityContext for the NextCloud pod, NextCloud container, nginx container, and the cronjob.

Based on #117 by @nlaille.

## Benefits

Allows users to set the securityContext for the NextCloud pod, NextCloud container, nginx container, and the cronjob.

## Possible drawbacks

Users can overwrite the default `runAs` property.

## Applicable issues

- fixes #186
- fixes #252

## Checklist 

- [x] DCO has been [signed off on the commit](https://docs.github.com/en/github/authenticating-to-github/signing-commits).
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] (optional) Variables are documented in the README.md

### CODE REVIEW

**Comment by Timmy1e** [comment]
@jessebot & @tvories, I've updated the PR, it's ready updated and for review.

**Comment by provokateurin** [comment]
Sorry I forgot to review this yesterday. I'll try to review it tomorrow if I don't forget it again :sweat_smile:

File: charts/nextcloud/templates/deployment.yaml [hunk 1]
```diff
@@ -121,10 +121,16 @@ spec:
...
         resources:
 {{ toYaml .Values.resources | indent 10 }}
+        {{- if .Values.nextcloud.securityContext}}
+        securityContext:
+          {{- with .Values.nextcloud.securityContext }}
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+        {{- end }}
+        volumeMounts:
+        {{- include "nextcloud.volumeMounts" . | trim | nindent 8 }}
```
Comment (provokateurin) [inline charts/nextcloud/templates/deployment.yaml:133]: Could you move this back above? I don't think there's a reason for this unrelated change

Comment (provokateurin) [inline charts/nextcloud/templates/deployment.yaml:133]: Could you move this back above? I don't think there's a reason for this unrelated change

File: charts/nextcloud/templates/deployment.yaml [hunk 2]
```diff
@@ -147,11 +153,17 @@ spec:
         {{-   end }}
         {{- end }}
         env:
-{{- include "nextcloud.env" . | indent 8 }}
+        {{- include "nextcloud.env" . | indent 8 }}
```
Comment (provokateurin) [inline charts/nextcloud/templates/deployment.yaml:156]: Also unrelated change

File: charts/nextcloud/templates/deployment.yaml [hunk 3]
```diff
@@ -147,11 +153,17 @@ spec:
...
 {{ toYaml .Values.resources | indent 10 }}
+        {{- if .Values.cronjob.securityContext}}
+        securityContext:
+          {{- with .Values.cronjob.securityContext }}
+            {{- toYaml . | nindent 10 }}
+          {{- end }}
+        {{- end }}
         volumeMounts:
-{{- include "nextcloud.volumeMounts" . | trim | nindent 8 }}
+        {{- include "nextcloud.volumeMounts" . | trim | nindent 8 }}
```
Comment (provokateurin) [inline charts/nextcloud/templates/deployment.yaml:166]: Also unrelated change

File: charts/nextcloud/templates/deployment.yaml [hunk 4]
```diff
@@ -308,23 +326,23 @@ spec:
...
+        # Will mount configuration files as www-data (id: 33) for nextcloud
         fsGroup: 33
-        {{- if .Values.securityContext }}
-        {{- with .Values.securityContext }}
-        {{- toYaml . | nindent 8 }}
         {{- end }}
+        {{- if .Values.securityContext }}
+          {{- with .Values.securityContext }}
+            {{- toYaml . | nindent 8 }}
+          {{- end }}
```
Comment (provokateurin) [inline charts/nextcloud/templates/deployment.yaml:345]: Since there is `.Values.nextcloud.podSecurityContext` now and `.Values.securityContext` is not documented anywhere, maybe just remove it?

**Review by provokateurin — CHANGES_REQUESTED** [review]

File: charts/nextcloud/templates/deployment.yaml (hunk 1 above)
Comment (Timmy1e) [reply charts/nextcloud/templates/deployment.yaml:133]: I moved it so all containers in this pod have `resources`, `securityContext`, `volumeMounts` in the same order.

File: charts/nextcloud/templates/deployment.yaml (hunk 4 above)
Comment (Timmy1e) [reply charts/nextcloud/templates/deployment.yaml:345]: I can remove it, but the reason I kept it in was for some backwards compatibility.

File: charts/nextcloud/templates/deployment.yaml (hunk 1 above)
Comment (provokateurin) [reply charts/nextcloud/templates/deployment.yaml:133]: Alright :)

File: charts/nextcloud/templates/deployment.yaml (hunk 4 above)
Comment (provokateurin) [reply charts/nextcloud/templates/deployment.yaml:345]: Yeah, but I wonder how many people actually rely on it since it's undocumented. Could you instead add a deprecation comment?

Comment (Timmy1e) [reply charts/nextcloud/templates/deployment.yaml:345]: I've added the deprecation comment to the `values.yaml` file.

**Comment by provokateurin** [comment]
Will review tomorrow :) I only work two days a week, so there is always a delay

**Review by provokateurin — APPROVED** [review]

**Comment by Timmy1e** [comment]
@jessebot Your assumption about `podSecurityContext` is correct, and it does appear that I have forgotten to update the description for `nextcloud.podSecurityContext`. Good catch! Feel free to open a PR if to fix it.

**Comment by provokateurin** [comment]
Run `helm template` to see the rendered output. This will probably point you to the source of the error.
